libs/libc/netdb/lib_dnsaddserver.c

```diff
@@ -198,10 +198,24 @@ int dns_add_nameserver(FAR const struct sockaddr *addr, socklen_t addrlen)
 {
   FAR uint16_t *pport;
   size_t copylen;
+  int nservers;
+  int idx;
 
   DEBUGASSERT(addr != NULL);
 
-  /* Copy the new server IP address into our private global data structure */
+  /* Get the index of the next free nameserver slot. */
+
+  dns_semtake();
+  if (g_dns_nservers == CONFIG_NETDB_DNSSERVER_NAMESERVERS)
```

Comment:
Should we pretend the new server to the beginning of the list? so we alwasy eliminate the oldest server and try the newest server in for each loop,

Reply:
Perhaps, but note that the resolv.conf version of the function appends the new nameserver at the end of the file. It should be changed to match for consistency. I would leave that to some future PR.

I think the most common use case is to get DNS1, DNS2,... IPs from operator and then add entire list by first calling dns_clear_servers() and then add the DNS1, DNS2, DNS3... and maybe add some generic fallback server like Google's 8.8.8.8 as *last* in case operator's preferred servers are down.

Reply:
> Perhaps, but note that the resolv.conf version of the function appends the new nameserver at the end of the file. It should be changed to match for consistency. I would leave that to some future PR.
> 

If so, it is better to append the new server to the end of list and iterate through the server backward.

> I think the most common use case is to get DNS1, DNS2,... IPs from operator and then add entire list by first calling dns_default_nameserver() and then add the DNS1, DNS2, DNS3... and maybe add some generic fallback server like Google's 8.8.8.8 as _last_ in case operator's preferred servers are down.